### PR TITLE
Updated docs

### PR DESCRIPTION
**Pull request by DavertMik**
Includes BDD documentation. And more to come.
Please review, leave comments. If you see mistakes - please fix them (it is easy to open and edit file and submit pull request)

### CODE REVIEW

File: docs/071-BDD.md [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+BDD has its own evolution from the days it was born, started by replacing "test" to "should" in unit tests, and moving towards powerful tools like Cucumber and Behat, which made user stories (human readable text) to be executed as an acceptance test. 
+
+The idea of story BDD can be narrowed to:
+
+* describe features in a scenario with a formal text
+* use examples to make abstract things concrete
+* implement each step of a scenario for testing
+* write actual code implementing the feature
+
+By making every feature to be formally written (in User Story format) and test automated, we ensure that business, developers, QAs, managers are in the same boat. That everyone in a team knows what is developed, what is not, what is tested and what is not. By making tests to be a part of user story BDD allows that non-technical personnel to write (or edit) acceptance tests. BDD also encourages exploration and discussions. To formalize the requirements and write user story in a way everyone can understand them, business and developers should talk, ask questions, and play the scenarios of features to implement.
```
Comment (javigomez) [inline docs/071-BDD.md]: ... and `play` the scenarios of features to implement.

I'm not sure to understand the word "play" in this sentence.

File: docs/071-BDD.md [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+Scenario:
+  Given I have product with 600 $ price in my cart 
+  And I have product with 1000 $ price
+  When I go to checkout process 
+  Then I should see that total number of products is 2
+  And my order amount is 1600 $
+```
+
+Cucucmber, Behat, and sure, **Codeception** can execute this scenario step by step as an automated test. 
+Every step in this scenario requires a code which defines .
```
Comment (javigomez) [inline docs/071-BDD.md]: `... requires a code which defines .`

I don't know how to write it better, but the current sentence doesn't sound correct to me.

**Comment by javigomez** [comment]
Sent updates #3054, #3053, #3052

**Comment by javigomez** [comment]
Added a doubt: #3072

**Comment by javigomez** [comment]
sent #3103

File: src/Codeception/Configuration.php [hunk 3]
```diff
@@ -85,7 +85,7 @@ class Configuration
         'settings'   => [
             'colors'     => false,
             'bootstrap'  => false,
-            'strict_xml' => false,
+            'strict_xml' => true,
```
Comment (DavertMik) [inline src/Codeception/Configuration.php]: accidentally added to commit. Anyway, this option does not affect anything

**Pull request merged**